editor/editor_run.cpp

```diff
@@ -131,29 +131,33 @@ Error EditorRun::run(const String &p_scene) {
 	}
 
 	switch (window_placement) {
+		// Coordinates are clamped to be always positive, as certain multi-monitor setups
+		// can cause them to become negative. However, negative window coordinates make no sense
+		// as window coordinates are absolute (and not relative to each monitor).
 		case 0: { // top left
+			const Vector2 pos = screen_rect.position.max(Vector2(0, 0));
```

Comment:
Negative `screen_rect` should not be possible in a first place, but even if it is, it should be fine to set window position to be it such screen rect. I guess the real issue is caused by the Linux/X11 `DisplayServer`s `set_window_position` and `screen_get_position` working in the different coordinate system for some reason.

Also, `0 x 0` point can be outside of any screen as well, I have display config like this, for example:
<img width="246" alt="disp" src="https://user-images.githubusercontent.com/7645683/154660448-142ed160-dc8b-4769-ba1a-e1d4f508f41d.png">

Reply:
I guess most likely `scene_rect` is not negative, but `pos:0 x 0, size:0x0` (can be the case if `XineramaQueryExtension` call failed) and negative values are caused by subtracting window size from the `0x0`.

Maybe it's better to check if `sceen_rect` size is zero, and simply remove `--position` from the command and rely on window manager to set a reasonable position.

Reply:
>> Just for added clarity I am not using Xinerama. My GPU's and XScreens are segregated to run separate workloads and screens.
> From #58178

And, `XineramaQueryExtension` failing seem to be expected outcome in this case.

Also, the same `screen_get_size(0) - window_size) / 2` is used in the multiple other places (in `Window` for popups, in `DisplayServers` for setting default main window position,  in `ProjectManager` and for editor scale auto-detection), these should check for invalid screen size as well.